### PR TITLE
Memory fixes

### PR DESCRIPTION
**Pull request by jserv**
Use clang-analyzer to figure out.

### CODE REVIEW

File: src/pm_memory_manager.c [hunk 1]
```diff
@@ -104,21 +107,23 @@ void* PMAlloc(PMMemoryManager* ctx, Class* klass)
...
   PMSlot_new(&prev_slot->next, pool, &ctx->pointer_map, prev_slot->size*2);
   slot = prev_slot->next;
   return PMSlot_alloc_obj(slot);
 }
     
 void* PMFree(PMMemoryManager* ctx, void* obj)
 {
-  op_assert((*(Class**)obj), "object %p is already freed\n", obj);
+  if (obj)
+    op_assert((*(Class**)obj), "object %p is already freed\n", obj);
```
Comment (dryman) [inline src/pm_memory_manager.c]: Please modify it to
```c
op_assert(obj, "Object pointer should not be NULL");
op_assert(((OPObject*)obj)->isa, "object %p is already freed\n", obj);
```
Thanks.

File: src/pm_memory_manager.c [hunk 2]
```diff
@@ -440,6 +447,7 @@ void* PMSlot_alloc_obj(PMSlot* self)
 
 void PMSlot_free_obj(PMSlot* self, void* obj)
 {
+  if (!obj) return;
```
Comment (dryman) [inline src/pm_memory_manager.c]: Please change it to assert.
```c
op_assert(obj, "Object pointer should not be NULL");
```
Thanks

File: src/pm_memory_manager.c [hunk 3]
```diff
@@ -276,12 +282,13 @@ PMMemoryManager* PMDeserialize(FILE* fd, ...)
       fread(&total_size, sizeof(size_t), 1, fd);
       log4c_category_log(de_logger, LOG4C_PRIORITY_INFO,
         "Deserializing class: %s", classname);
-      free(classname);
 
       Class* klass = LPTypeMap_get(classname);
       log4c_category_log(de_logger, LOG4C_PRIORITY_INFO,
         "Deserializing, found matching klass addr: %p, %s",
         klass, klass->classname);
+      free(classname);
```
Comment (dryman) [inline src/pm_memory_manager.c]: Nice catch, thanks.

File: src/pm_memory_manager.c (hunk 1 above)
Comment (jserv) [inline src/pm_memory_manager.c]: Done!

File: src/pm_memory_manager.c (hunk 2 above)
Comment (jserv) [inline src/pm_memory_manager.c]: Done.

**Comment by jserv** [comment]
ping?

**Comment by dryman** [comment]
Closing old PR.

**Pull request closed without merging**